### DISCUSSION TITLE
v3.8.0

### DISCUSSION THREAD

**Discussion by markwhitfeld** [ngxs/store#1985] [Releases]
## Important Notes:
- In this release we are publishing in the new Ivy packaging format, and therefore have dropped support for Angular 11 and earlier.
- You may get some **TypeScript compile failures** after you update... but this is not a regression. The selectors and state operators now have really solid typings, so this would most likely identify issues in your code that you weren't aware of :heart: .

## What's Changed
* feat(store): strongly type ofAction* methods by @bbarry in https://github.com/ngxs/store/pull/1808
* feat(store): expose ActionContext and ActionStatus by @arturovt in https://github.com/ngxs/store/pull/1766
* feat(store): warn on unhandled actions by @arturovt in https://github.com/ngxs/store/pull/1870
* feat(store): allow prodiving `warnOnUnhandledActions` as boolean by @arturovt in https://github.com/ngxs/store/pull/1951
* feat(store): build packages in Ivy format by @arturovt in https://github.com/ngxs/store/pull/1945
* feat(store): improve create selector types by @markwhitfeld in https://github.com/ngxs/store/pull/1982
* feat(operators): improve publicly exposed types by @markwhitfeld in https://github.com/ngxs/store/pull/1956
* feat(operators): improve type checking and intellisense for State Operators by @markwhitfeld in https://github.com/ngxs/store/pull/1947
* feat(selector-utils): add advanced selector utils by @Carniatto in https://github.com/ngxs/store/pull/1824
* perf(store): tree-shake patch errors by @arturovt in https://github.com/ngxs/store/pull/1955
* fix(store): do not infer T from argument to patch by @david-shortman in https://github.com/ngxs/store/pull/1806
* fix(store): get descriptor explicitly when considered as class prop by @arturovt in https://github.com/ngxs/store/pull/1961
* fix(store): avoid delayed updates from state stream by @arturovt in https://github.com/ngxs/store/pull/1981
* feat(storage-plugin): allow providing namespace for keys by @arturovt in https://github.com/ngxs/store/pull/1841
* feat(storage-plugin): enable providing storage engine individually by @arturovt in https://github.com/ngxs/store/pull/1935
* feat(router-plugin): provide more actions and navigation timing option by @arturovt in https://github.com/ngxs/store/pull/1932
* feat(form-plugin): allow `ngxsFormDebounce` to be string by @arturovt in https://github.com/ngxs/store/pull/1972
* feat(devtools-plugin): add trace options to `NgxsDevtoolsOptions` by @Dyqmin in https://github.com/ngxs/store/pull/1968
* feat(devtools-plugin): add new options to the `NgxsDevtoolsOptions` interface by @oleksandr-codefresh in https://github.com/ngxs/store/pull/1879
* chore(deps): update dependency express to v4.17.3 [security] by @renovate in https://github.com/ngxs/store/pull/1949
* test(unit): update unit test setup, revise existing unit tests by @rfprod in https://github.com/ngxs/store/pull/1846
* docs: update action-handlers.md by @maxaxehax in https://github.com/ngxs/store/pull/1973

## New Contributors
* @oleksandr-codefresh made their first contribution in https://github.com/ngxs/store/pull/1879
* @Dyqmin made their first contribution in https://github.com/ngxs/store/pull/1968
* @maxaxehax made their first contribution in https://github.com/ngxs/store/pull/1973
* @david-shortman made their first contribution in https://github.com/ngxs/store/pull/1806
* @bbarry made their first contribution in https://github.com/ngxs/store/pull/1808

**Full Changelog**: https://github.com/ngxs/store/compare/v3.7.6...v3.8.0

<hr /><em>This discussion was created from the release <a href='https://github.com/ngxs/store/releases/tag/v3.8.0'>v3.8.0</a>.</em>

**Comment by markwhitfeld** [comment]
If you have successfully updated, please add a 👍 reaction to this comment. If your update PR is public and includes interesting changes around the typing improvements, please add a comment in this answer's thread with a link to the PR.
This could be useful to others in the community that need to tweak types as a result of the improved typing in this version.
If you have interesting changes and your codebase is private (so you can't share the PR), then you are welcome to share a screenshot or snippet of the typings fixes that you had to do.

**Reply by vmuresanu** [reply]
Thanks to @markwhitfeld  and his reply, I was able to migrate to v.3.8.0 and resolve my issue. In order to see what changes I faced during the migration, please check this comment - https://github.com/ngxs/store/issues/1986#issuecomment-1491640969
